### PR TITLE
Towards Quality Level 1 - Added common_linters and fixed tests 

### PR DESCRIPTION
**Pull request by ahcorde**
- Added common_linters
 - Fixed C++ tests
 - Fixed Python API
 - Fixed Python tests
 - Fixed license headers

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: resource_retriever/CMakeLists.txt
Comment (clalancette) [inline resource_retriever/CMakeLists.txt]: Why are we installing the tests?  I don't think this is something we want in the "released" binaries.

File: resource_retriever/test/CMakeLists.txt
Comment (clalancette) [inline resource_retriever/test/CMakeLists.txt]: The top-level CMakeLists.txt has no `include` or `add_subdirectory` directives that I can see.  How is this getting called and built?

File: resource_retriever/test/test.cpp
Comment (clalancette) [inline resource_retriever/test/test.cpp]: I think gtest has a `FAIL_STREAM` or something like that, so you can combine these two lines.

File: resource_retriever/test/test.cpp
Comment (clalancette) [inline resource_retriever/test/test.cpp]: I know you didn't add this, but putting a message in here on why it failed would be helpful (probably using `e.what()`).  Same goes for all of the other empty `FAIL` below.

File: resource_retriever/test/test.cpp
Comment (clalancette) [inline resource_retriever/test/test.cpp]:
```suggestion
  } catch (const std::exception & e) {
```

(same for all of the other exceptions below)

File: resource_retriever/test/test.cpp
Comment (clalancette) [inline resource_retriever/test/test.cpp]: I don't think we need to print anything here; the test case succeeded.  Same goes for all of the other tests below.

**Review by clalancette — CHANGES_REQUESTED** [review]
A few things to fix in here, and some questions.

File: resource_retriever/CMakeLists.txt
Comment (ahcorde) [reply resource_retriever/CMakeLists.txt]: The test is looking for a specific file https://github.com/ros/resource_retriever/blob/a0b44a592dc55408f5515a07900a39eced41724d/resource_retriever/test/test.cpp#L40

I think this should be installed otherwise the code will not find it

File: resource_retriever/test/CMakeLists.txt
Comment (ahcorde) [reply resource_retriever/test/CMakeLists.txt]: I forgot to remove it https://github.com/ros/resource_retriever/pull/43/commits/edd0621053d8eb60d5145d1caab15542a7bd583a 

I moved all the logic to the CMakeLists.txt like ROS2 packages

File: resource_retriever/test/test.cpp
Comment (ahcorde) [reply resource_retriever/test/test.cpp]: https://github.com/ros/resource_retriever/pull/43/commits/071a03377748758054f7ac8b2c13451678a889dd

File: resource_retriever/test/test.cpp
Comment (ahcorde) [reply resource_retriever/test/test.cpp]: https://github.com/ros/resource_retriever/pull/43/commits/3b5aa92912f0f4156bcada7d36cf80a201323ccc

File: resource_retriever/test/test.cpp
Comment (ahcorde) [reply resource_retriever/test/test.cpp]: https://github.com/ros/resource_retriever/pull/43/commits/09863dbc1aa7749bb34bc129c98dd4e2a3f0af17

File: resource_retriever/test/test.cpp
Comment (ahcorde) [reply resource_retriever/test/test.cpp]: https://github.com/ros/resource_retriever/pull/43/commits/d8e3f947d82e15daa615b4d00c6e7374ae5fd906

File: resource_retriever/CMakeLists.txt
Comment (clalancette) [reply resource_retriever/CMakeLists.txt]: While that is true, it also means that the tests will all make it into a binary once we build it.  I don't think that is desirable.

I'm not quite sure how to do this and still test that this is looking things up in the package share directory.  Any other thoughts on how to do this?  At the very least, we should only install `test.txt`, not all of the `test` subdirectory.

Comment (ahcorde) [reply resource_retriever/CMakeLists.txt]: https://github.com/ros/resource_retriever/pull/43/commits/63fc6e7cca798dd894508c6dbc031f25138a55a8

Comment (clalancette) [reply resource_retriever/CMakeLists.txt]: Thanks, that's a bit better.  @sloretz Any other thoughts here?

**Review by clalancette — APPROVED** [review]
This looks OK to me now, but I'd like to get a review from @sloretz before merging.

File: resource_retriever/CMakeLists.txt
Comment (sloretz) [inline resource_retriever/CMakeLists.txt]: Nit, no need to check for `TARGET ${PROJECT_NAME}_test`'s existence. If `BUILD_TESTING` was true above, then the test target better exist.

**Review by sloretz — COMMENTED** [review]
LGTM with one nit and green CI

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10335/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5863/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8421/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10244/)

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10374/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5889/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8450/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10270/)

**Comment by ahcorde** [comment]
Fixing warning on windows
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10272/)

**Pull request merged**
